File: skypydb/api/mixins/vector/delete_collection.py
"""Mixin that deletes vector collections for `VectorClient`."""


class DeleteCollectionMixin:
    """Expose collection deletion behavior on `VectorClient`."""

    def delete_collection(self, name: str) -> None:
        """Delete a collection and remove any cached wrapper."""

        self._db.delete_collection(name)
        if name in self._collections:
            del self._collections[name]

File: skypydb-js/src/api/operations/vector/delete_collection.ts
import { ListCollectionsMixin } from "./list_collections";
import type { VectorDatabase } from "../../../database/vector_database";

export class DeleteCollectionMixin extends ListCollectionsMixin {
  async delete_collection(name: string): Promise<void> {
    this._db.delete_collection(name);
    this._collections.delete(name);
  }
}
